python/pyiceberg/catalog/dynamodb.py

```diff
@@ -0,0 +1,776 @@
+
+        try:
+            self.drop_table(from_identifier)
+        except (NoSuchTableError, GenericDynamoDbError) as e:
+            self.drop_table(to_identifier)
+            raise ValueError(
+                f"Failed to drop old table {from_database_name}.{from_table_name}, "
+                f"after renaming to {to_database_name}.{to_table_name}. Rolling back to use the old one."
+            ) from e
+
+        return self.load_table(to_identifier)
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Properties = EMPTY_DICT) -> None:
+        """Create a namespace in the catalog.
+
+        Args:
+            namespace: Namespace identifier
+            properties: A string dictionary of properties for the given namespace
+
+        Raises:
+            ValueError: If the identifier is invalid
+            AlreadyExistsError: If a namespace with the given name already exists
+        """
+        database_name = self.identifier_to_database(namespace)
+
+        try:
+            self._put_dynamo_item(
+                item=_get_create_database_item(database_name=database_name, properties=properties),
+                condition_expression=f"attribute_not_exists({DYNAMODB_COL_NAMESPACE})",
+            )
+        except ConditionalCheckFailedException as e:
+            raise NamespaceAlreadyExistsError(f"Database {database_name} already exists") from e
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        """Drop a namespace.
+
+        A Glue namespace can only be dropped if it is empty
+
+        Args:
+            namespace: Namespace identifier
+
+        Raises:
+            NoSuchNamespaceError: If a namespace with the given name does not exist, or the identifier is invalid
+            NamespaceNotEmptyError: If the namespace is not empty
+        """
+        database_name = self.identifier_to_database(namespace, NoSuchNamespaceError)
+        table_identifiers = self.list_tables(namespace=database_name)
+
+        if len(table_identifiers) > 0:
+            raise NamespaceNotEmptyError(f"Database {database_name} is not empty")
+
+        try:
+            self.dynamodb.delete_item(
+                TableName=self.dynamodb_table_name,
+                Key={
+                    DYNAMODB_COL_IDENTIFIER: {
+                        "S": DYNAMODB_NAMESPACE,
+                    },
+                    DYNAMODB_COL_NAMESPACE: {
+                        "S": database_name,
+                    },
+                },
+                ConditionExpression=f"attribute_exists({DYNAMODB_COL_NAMESPACE})",
+            )
+        except self.dynamodb.exceptions.ConditionalCheckFailedException as e:
+            raise NoSuchNamespaceError(f"Database does not exist: {database_name}") from e
+        except (
+            self.dynamodb.exceptions.ProvisionedThroughputExceededException,
```

Comment:
Should we generalize these blocks? Since we're catching the same exceptions all the time. WDYT?

Reply:
Done.

Reply:
I tried to doa function, also a variable:
```
ddb_put_and_delete_exceptions: Tuple[Exception, ...] = (
            self.dynamodb.exceptions.ResourceNotFoundException,
            self.dynamodb.exceptions.ItemCollectionSizeLimitExceededException,
            self.dynamodb.exceptions.TransactionConflictException,
        )
```

But mypy fails me:
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

python/pyiceberg/catalog/dynamodb.py:358: error: Exception type must be derived from BaseException  [misc]
...
```

I think it's not compatible somehow. I reverting this change. But please let me know if there is a good way to do this.